src/node/socket.ts

```diff
@@ -76,7 +75,10 @@ export class SocketProxyProvider {
       this._proxyServer = this.findFreeSocketPath(this.proxyPipe)
         .then((pipe) => {
           this.proxyPipe = pipe
-          return Promise.all([fs.mkdir(tmpdir, { recursive: true }), fs.rmdir(this.proxyPipe, { recursive: true })])
+          return Promise.all([
+            fs.mkdir(paths.runtime, { recursive: true }),
+            fs.rmdir(this.proxyPipe, { recursive: true }),
+          ])
```

Comment:
I was a bit confused looking at this code. Previously, we created a directory with `tmpdir` and removed the `proxyPipe`.

In my change, I have us create a directory at `paths.runtime`.

Reply:
Ah yup this looks good. We are making sure the directory the socket will go into exists and at the same time removing the old socket file if it exists.

Reply:
👍

Reply:
💝

Reply:
👍

Reply:
👍

Reply:
👌

Reply:
> I was a bit confused looking at this code. Previously, we created a directory with `tmpdir` and removed the `proxyPipe`.
> 
> In my change, I have us create a directory at `paths.runtime`.